crates/re_query_cache/src/cache.rs

```diff
@@ -122,34 +141,37 @@ impl Caches {
         A: Archetype,
         F: FnMut(&mut LatestAtCache) -> R,
     {
-        let key = CacheKey::new(store_id, entity_path, query.timeline);
-
-        let cache =
-            re_data_store::DataStore::with_subscriber_once(*CACHES, move |caches: &Caches| {
-                let mut caches = caches.0.write();
-
-                let caches_per_archetype = caches.entry(key.clone()).or_default();
-
-                let removed_bytes = caches_per_archetype.handle_pending_invalidation();
-                if removed_bytes > 0 {
-                    re_log::trace!(
-                        store_id = %key.store_id,
-                        entity_path = %key.entity_path,
-                        removed = removed_bytes,
-                        "invalidated latest-at caches"
-                    );
-                }
-
-                let mut latest_at_per_archetype =
-                    caches_per_archetype.latest_at_per_archetype.write();
-                let latest_at_cache = latest_at_per_archetype.entry(A::name()).or_default();
-
-                Arc::clone(latest_at_cache)
+        assert!(
+            self.store_id == *store.id(),
+            "attempted to use a query cache {} with the wrong datastore ({})",
+            self.store_id,
+            store.id(),
+        );
```

Comment:
I argue this should always be a hard crash; this is just so fundamentally wrong.

Not sure how you would ever get into that situation to begin with either, you'd have to use the store from one `StoreHub` and the cache from another...